rules/naming/src/ValueObject/PropertyRename.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Naming\ValueObject;
+
+use PhpParser\Node\Stmt\ClassLike;
+use PhpParser\Node\Stmt\Property;
+use PHPStan\Type\Type;
+
+final class PropertyRename extends Property
+{
+    /**
+     * @var string
+     */
+    private $expectedName;
+
+    /**
+     * @var string
+     */
+    private $currentName;
+
+    /**
+     * @var Property
+     */
+    private $property;
+
+    /**
+     * @var Type
+     */
+    private $propertyType;
+
+    /**
+     * @var ?ClassLike
+     */
+    private $classLike;
+
+    public function __construct(
+        Property $property,
+        string $expectedName,
+        string $currentName,
+        Type $type,
+        ?ClassLike $classLike
+    ) {
+        $this->property = $property;
+        $this->expectedName = $expectedName;
+        $this->currentName = $currentName;
+        $this->propertyType = $type;
+        $this->classLike = $classLike;
+    }
+
+    public function getProperty(): Property
+    {
+        return $this->property;
+    }
+
+    public function getExpectedName(): string
+    {
+        return $this->expectedName;
+    }
+
+    public function getCurrentName(): string
+    {
+        return $this->currentName;
+    }
+
+    public function getPropertyType(): Type
+    {
+        return $this->propertyType;
+    }
+
+    public function getClassLike(): ?ClassLike
+    {
+        return $this->classLike;
+    }
```

Comment:
In what case the property can be without Class_?

Reply:
`$property->getAttribute(AttributeKey::CLASS_NODE)` can return null and in case, for some reason, it will be null it will skip renaming in PropertyFetch 
https://github.com/rectorphp/rector/pull/4242/files#diff-397302fe4eb22d04e3248d9f60054cbcR53-R55

Hm... I'll make it non-nullable and add docblock that it is just a ClassLike...

Reply:
If it's not null on purpose, maybe it should be skipped ealier before full rename.

Also, throwing `ShouldNotHappenException` makes sense here.

Reply:
> make it non-nullable

:+1:

Reply:
Done